src/autopas/iterators/ParticleIterator.h

```diff
@@ -91,45 +105,82 @@ class ParticleIterator : public ParticleIteratorInterfaceImpl<Particle, modifiab
           "nullptr!");
     }
 
-    if (not _iteratorWithinOneCell.isValid() or not isCellTypeBehaviorCorrect()) {
-      ParticleIterator::next_non_empty_cell();
-    }
-    // The iterator might still be invalid (because the cell is empty or the owned-state of the particle is wrong), so
-    // we check it here!
-    if (not ParticleIterator::isValid() and this->_iteratorAcrossCells < cont->end()) {
-      // the iterator is invalid + we still have particles/cells, so we increment it!
-      ParticleIterator::operator++();
+    if (_additionalParticleVectorToIterateState != AdditionalParticleVectorToIterateState::iterating) {
+      if (not _iteratorWithinOneCell.isValid() or not isCellTypeBehaviorCorrect()) {
+        // This ensures that the current cell is NOT empty or (that the end of the cells is selected and
+        // ParticleIterator::isValid is false).
+        /// @todo: this might not actually be needed -> check!
+        ParticleIterator::next_non_empty_cell();
```

Comment:
```suggestion
        this->next_non_empty_cell();
```
not sure about syntax, especially since we are in a constructor, but can't you use `this` here? The way you wrote it looks so `static`.

Reply:
in a constructor you shouldn't use this-> on virtual functions ;)